src/rust/ide/enso-protocol/src/language_server/types.rs

```diff
@@ -37,6 +37,14 @@ impl Display for Path {
 }
 
 impl Path {
+    /// Splits path into name and segments. e.g.:
+    /// Path{root_id,segments:["foo","bar","qux"]} => ("qux",Path{root_id,segments:["foo","bar"]})
+    pub fn split_into_name_and_segments(&self) -> (String,Path) {
```

Comment:
I'd name rather like `split_into_file_name_and_parent` — the point is that new segments are the parent path. After all, we had segments all along.